slyd/slyd/utils/storage.py

```diff
@@ -1,34 +1,114 @@
+import errno
 import os
+import os.path
+import re
 import shutil
 
+from collections import OrderedDict
+
+from django.conf import settings
 from django.core.files.base import ContentFile
 from django.core.files.move import file_move_safe
-from django.core.files.storage import Storage, FileSystemStorage
+from django.core.files.storage import FileSystemStorage, Storage
 from dulwich.diff_tree import tree_changes
-from dulwich.objects import Blob
+from dulwich.objects import Blob, Tree
+from six import iteritems, text_type, string_types
+
+from .projecttemplates import templates
+from .repoman import Repoman, DEFAULT_USER, FILE_MODE
 
 
-FILE_MODE = 0o100644
+class InvalidFilename(Exception):
+    pass
 
 
-class CommitingStorage(object):
+class CommittingStorage(object):
```

Comment:
I think this object is not getting us any benefit here. Can we move these methods to `BasePortiaStorage` as all storage classes will inherit from this?